packages/grafana-ui/src/components/Splitter/Splitter.tsx

```diff
@@ -0,0 +1,482 @@
+import { css } from '@emotion/css';
+import { clamp, throttle } from 'lodash';
+import React, { useCallback, useId, useLayoutEffect, useRef } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+import { DragHandlePosition, getDragStyles } from '../DragHandle/DragHandle';
+
+export interface Props {
+  /**
+   * The initial size of the primary pane between 0-1, defaults to 0.5
+   */
+  initialSize?: number;
+  direction?: 'row' | 'column';
+  dragPosition?: DragHandlePosition;
+  primaryPaneStyles?: React.CSSProperties;
+  secondaryPaneStyles?: React.CSSProperties;
+  /**
+   * Called when ever the size of the primary pane changes
+   * @param size (float from 0-1)
+   */
+  onSizeChange?: (size: number) => void;
+  children: [React.ReactNode, React.ReactNode];
+}
+
+/**
+ * Splits two children into two resizable panes
+ * @alpha
+ */
+export function Splitter(props: Props) {
+  const {
+    direction = 'row',
+    initialSize = 0.5,
+    primaryPaneStyles,
+    secondaryPaneStyles,
+    onSizeChange,
+    dragPosition = 'middle',
+    children,
+  } = props;
+
+  const {
+    kids,
+    containerRef,
+    firstPaneRef,
+    minDimProp,
+    splitterRef,
+    measurementProp,
+    handleSize,
+    onPointerUp,
+    onPointerDown,
+    onPointerMove,
+    onKeyDown,
+    onKeyUp,
+    onDoubleClick,
+    onBlur,
+    secondPaneRef,
```

Comment:
this is a lot of returned "things". could we instead group some of them into one thing like `splitterProps` that can then be spread onto the splitter element? i'm mostly thinking about the interaction stuff (`onPointer`, `onKey`, `onBlur`, etc.), since they all have to go on the same element anyway

Reply:
you could also return the `tabIndex`, associated `aria-value*` attributes and `role` stuff as part of the `splitterProps` as well if you pass through the `initialSize`? that way you wouldn't have to manipulate the dom directly in the hook to change `ariaValueNow`

Reply:
@ashharrison90 I moved most to a splitterProps and got rid of a few others, I did not like adding role, aria props tabIndex etc,  because then I basically have move everything to the hook and there i no nothing in the markup generation left so the separation between callback / static logic and markup rendering becomes lost. 

I agree if useSplitter was designed to be used outside this component is should probably return all props, but here it just a way to separate all the callbacks from the rendering